src/libraries/System.Reflection.Emit/tests/PersistableAssemblyBuilder/AssemblySaveTestsWithVariousTypes.cs

```diff
@@ -0,0 +1,455 @@
+
+                AssemblyTools.WriteAssemblyToDisk(s_assemblyName, types, file.Path);
+
+                Assembly assemblyFromDisk = AssemblyTools.TryLoadAssembly(file.Path);
+
+                Assert.NotNull(assemblyFromDisk);
+
+                Module moduleFromDisk = assemblyFromDisk.Modules.First();
+                Type[] typesFromDisk = moduleFromDisk.GetTypes();
+
+                Assert.NotNull(moduleFromDisk);
+                Assert.Equal(1, typesFromDisk.Length);
+
+                Type sourceType = types[0];
+                Type typeFromDisk = typesFromDisk[0];
+
+                Assert.Equal(sourceType.Name, typeFromDisk.Name);
+                Assert.Equal(sourceType.Namespace, typeFromDisk.Namespace);
+                Assert.Equal(sourceType.Attributes, typeFromDisk.Attributes);
+
+                Assert.Empty(typeFromDisk.GetMethods());
+            }
+        }
+
+        [Fact]
+        public void TwoEmptyInterfaces()
+        {
+            using (TempFile file = TempFile.Create())
+            {
+                Type[] types = new Type[] { typeof(INoMethod), typeof(INoMethod2) };
+
+                AssemblyTools.WriteAssemblyToDisk(s_assemblyName, types, file.Path);
+
+                Assembly assemblyFromDisk = AssemblyTools.TryLoadAssembly(file.Path);
+
+                Assert.NotNull(assemblyFromDisk);
+
+                Module moduleFromDisk = assemblyFromDisk.Modules.First();
+                Type[] typesFromDisk = moduleFromDisk.GetTypes();
+
+                Assert.NotNull(moduleFromDisk);
+                Assert.Equal(2, typesFromDisk.Length);
+
+                // Type comparisons
+                for (int i = 0; i < types.Length; i++)
+                {
+                    Type sourceType = types[i];
+                    Type typeFromDisk = typesFromDisk[i];
+
+                    Assert.Equal(sourceType.Name, typeFromDisk.Name);
+                    Assert.Equal(sourceType.Namespace, typeFromDisk.Namespace);
+                    Assert.Equal(sourceType.Attributes, typeFromDisk.Attributes);
+
+                    Assert.Empty(typeFromDisk.GetMethods());
+                }
+            }
+        }
+
+        [Fact]
+        public void TwoInterfaceOneMethod()
+        {
+            using (TempFile file = TempFile.Create())
+            {
+                Type[] types = new Type[] { typeof(INoMethod), typeof(IOneMethod) };
+
+                AssemblyTools.WriteAssemblyToDisk(s_assemblyName, types, file.Path);
+
+                Assembly assemblyFromDisk = AssemblyTools.TryLoadAssembly(file.Path);
+
+                Assert.NotNull(assemblyFromDisk);
+
+                Module moduleFromDisk = assemblyFromDisk.Modules.First();
+                Type[] typesFromDisk = moduleFromDisk.GetTypes();
+
+                Assert.NotNull(moduleFromDisk);
+                Assert.Equal(2, typesFromDisk.Length);
+
+                // Type comparisons
+                for (int i = 0; i < types.Length; i++)
```

Comment:
Since there are only two types, just duplicate the tests? That will also avoid ambiguity with the `<= 1` and `sourceMethods.Length > 0` cases below. Add some nested methods if sharing is desired.

Also applies elsewhere.